### ISSUE TITLE
Collection Time of per iteration

### ISSUE THREAD

**Issue by ChengEeee** [chengxuxin/extreme-parkour#22]
![image](https://github.com/chengxuxin/extreme-parkour/assets/94172651/ce59beea-245f-4589-9675-bf98d6927329)
Why does the  project's Collection Time of per iteration take several times more time than the LeggedGym's.
Hope your reply！Thank you very much

**Comment by ChengEeee** [comment]
LeggedGym's Collection Time of per iteration less than 1s.
and i have closed the draw goal points and the heightmap

**Comment by 1242713693** [comment]
Hello, is your display graphical interface in VScode? My computer doesn't have a GPU, can I run it directly on the CPU? If you run directly on the CPU, can you visualize, like in your video! thank you.

**Comment by chengxuxin** [comment]
> ![image](https://private-user-images.githubusercontent.com/94172651/295166283-ce59beea-245f-4589-9675-bf98d6927329.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxMDUxNTcsIm5iZiI6MTcwOTEwNDg1NywicGF0aCI6Ii85NDE3MjY1MS8yOTUxNjYyODMtY2U1OWJlZWEtMjQ1Zi00NTg5LTk2NzUtYmY5OGQ2OTI3MzI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDA3MjA1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkYWUxZWZkY2RiYjc1MmI1YWYwMmMzYTJlYmFlNGFmMWUyNmYxZmI0NjRjOGNlYjlmYzc5YjU4OTQwODUyMWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._hiRt9nnOwYh-o0SDNQnAAPLOkbLj6-e9VeR-1_Sfm0) Why does the project's Collection Time of per iteration take several times more time than the LeggedGym's. Hope your reply！Thank you very much

Please provide more info on what commands and gpu you are using.

**Comment by chengxuxin** [comment]
> Hello, is your display graphical interface in VScode? My computer doesn't have a GPU, can I run it directly on the CPU? If you run directly on the CPU, can you visualize, like in your video! thank you.

Yes you can train teacher and student without gui. To visualize on a headless machine please add `--web` arg to visualize in vscode.

**Comment by 1242713693** [comment]
> > Hello, is your display graphical interface in VScode? My computer doesn't have a GPU, can I run it directly on the CPU? If you run directly on the CPU, can you visualize, like in your video! thank you.
> 
> Yes you can train teacher and student without gui. To visualize on a headless machine please add `--web` arg to visualize in vscode.

I have installed Live Preview in VScode according to the prompt, and the code can run normally, but there is no interface. I do not know whether it is my wrong operation or other problems. I sincerely hope to get your help, thank you
![2024-02-28 15-49-16屏幕截图](https://github.com/chengxuxin/extreme-parkour/assets/133180305/649f8390-f771-4cd7-8515-f8425b87fb87)

**Comment by chengxuxin** [comment]
The `train.py` won't give you rendering. You have to run `python play.py --exptid the_model_you_trained --web` to get a render of your policy. Please see the readme carefully.

**Comment by 1242713693** [comment]
> The `train.py` won't give you rendering. You have to run `python play.py --exptid the_model_you_trained --web` to get a render of your policy. Please see the readme carefully.

Thank you very much for your help. Do you mean that I can only watch in vscode through the command --web after the training is completed? If the model is being trained, can it also be viewed via --web?
There is another thing that needs to trouble you, if I configure the GPU for the computer, basically, which model of the ordinary Nvidia graphics card can make the code run smoothly? thank you.

**Comment by chengxuxin** [comment]
you can run `play` while you are running `train`, no problem. I recommend at least 16 gb gpu memory, best would be 3090 4090.
